src/core/drive/visit.js

```diff
@@ -335,7 +335,7 @@ export class Visit {
   // Scrolling
 
   performScroll() {
-    if (!this.scrolled && !this.view.forceReloaded) {
+    if (!this.scrolled && !this.view.forceReloaded && !this.view.snapshot.shouldPreserveScrollPosition) {
```

Comment:
Since `shouldPreserveScrollPosition` doesn't check if the page is a refresh, this restores scroll even when doing a classic visit instead of a refresh. It was unexpected to me when I tried it out, we should only restore scroll for a refresh, shouldn't we ?

Reply:
This is intentional. We have two different settings `<meta name="turbo-refresh-method">` and `<meta name="turbo-refresh-scroll">` because we have identified some use cases in our apps where we want to do one without the other necessarily.

Reply:
@afcapel is scroll preservation something that's specific to Refresh Morph?

Could scroll preservation be implemented separately, with a more generic `<meta>` name?

Maybe something like `<meta name="turbo-scroll" value="preserve">`? The scroll-related portion of the diff feels small by comparison. I'd be happy to open a PR to add that behavior separately.

Reply:
I've opened https://github.com/hotwired/turbo/pull/1040 to add `<meta name="turbo-scroll" content="preserve">` support. That PR is against `main`.

I think there's a lot of value in managing scroll preservation outside of a morphing context, so generalizing to `turbo-scroll` feels like a win.

Reply:
@seanpdoyle I'd like to keep the scroll configuration under the same "page refresh" umbrella. Our vision here is not adding more flexibility but aiming for seamless smoother page updates (as in re-render the current page again). We believe it's in that scenario where keeping the scroll makes sense, to the point where "preserve" should be the future default for such cases being "reset" the exception. In fact, the only reason we added the option is that we found a case of "page refresh" where you wanted to opt out of scroll preservation due to design considerations.

Just like morphing, scroll preservation for regular navigation could make sense in other scenarios, but we believe those will be quite rare and are out of the scope of what we're addressing here. I'd like to keep both "scroll" and "morph" tied to "page refreshes" for now, as I think it makes for a more cohesive development story.